### ISSUE TITLE
Double render when using SeatsioDesigner

### ISSUE THREAD

**Issue by MiguiHidalgo** [seatsio/seatsio-react#51]
Im trying to implement the ``<SeatsioDesigner/>`` on a Vite.js React project and the result is the component renders twice.

```
<div id='seats-io' style={{ 'height': '500px' }}>
                       <SeatsioDesigner
                            divId='seats-io'
                            onExitRequested={false}
                            secretKey='my_secret'
                            region='eu'
                            mode='normal' 
                        />
                    </div>
```
![](https://cdn.discordapp.com/attachments/634716470370238476/1026489082810998855/Screenshot_2022-10-03_at_15.38.01.png)

![](https://cdn.discordapp.com/attachments/634716470370238476/1026490774176673902/Screenshot_2022-10-03_at_15.47.19.png)

**Comment by mroloux** [comment]
Hi,

Do you have some pointers on how to reproduce this? So far we couldn't.

Thanks

**Comment by MiguiHidalgo** [comment]
> This has something todo with how react detects unexpected side effects. If you have **React Strict mode on** and **you are in development mode** a lot of lifecycle hooks are just called two times (https://reactjs.org/docs/strict-mode.html#detecting-unexpected-side-effects). **In production this will not happen!!**
> 
> In development react will call `componentDidMount` twice:
> 
> https://github.com/seatsio/seatsio-react/blob/fbd693470ab845f94e21960b02497e695a379746/src/main/Embeddable.js#L12-L14
> 
> And because `createAndRenderChart` does not check if there was already a chart created it will just create two.
> 
> https://github.com/seatsio/seatsio-react/blob/fbd693470ab845f94e21960b02497e695a379746/src/main/Embeddable.js#L23-L31
> 
> Without looking further into the code I would suspect that something like this would already be sufficient:
> 
> ```js
> // This is not tested
> if(!this.chart) this.chart = this.createChart(seatsio, config).render() 
> ```

Yes it was related to Rect Strict mode

**Comment by mroloux** [comment]
Thanks for the input. This issue should be fixed in v13.1.
